crates/router/src/types/domain/behaviour.rs

```diff
@@ -1,32 +1 @@
-use common_utils::errors::{CustomResult, ValidationError};
-
-use crate::pii::Secret;
-
-/// Trait for converting domain types to storage models
-#[async_trait::async_trait]
-pub trait Conversion {
-    type DstType;
-    type NewDstType;
-    async fn convert(self) -> CustomResult<Self::DstType, ValidationError>;
-
-    async fn convert_back(
-        item: Self::DstType,
-        key: &Secret<Vec<u8>>,
-    ) -> CustomResult<Self, ValidationError>
-    where
-        Self: Sized;
-
-    async fn construct_new(self) -> CustomResult<Self::NewDstType, ValidationError>;
-}
-
-#[async_trait::async_trait]
-pub trait ReverseConversion<SrcType: Conversion> {
-    async fn convert(self, key: &Secret<Vec<u8>>) -> CustomResult<SrcType, ValidationError>;
-}
-
-#[async_trait::async_trait]
-impl<T: Send, U: Conversion<DstType = T>> ReverseConversion<U> for T {
-    async fn convert(self, key: &Secret<Vec<u8>>) -> CustomResult<U, ValidationError> {
-        U::convert_back(self, key).await
-    }
-}
+pub use hyperswitch_domain_models::behaviour::{Conversion, ReverseConversion};
```

Comment:
Can we move this as an inline module within the parent module (`domain.rs`)?

```rust
mod behaviour {
    pub use hyperswitch_domain_models::behaviour::{Conversion, ReverseConversion};
}
```

Reply:
can you please explain why this change would be necessary?

Reply:
Not a necessary change, only an improvement that we can take up. My reasoning is that we're creating individual files for modules like this which only re-export items from somewhere else and have no other code. I think these can be moved to inline modules instead.

Reply:
Got it